stdlib/strings.ncl

```diff
@@ -51,43 +121,305 @@
+
+      For example:
+      ```nickel
+        match "5 apples, 6 pears and 0 grapes" "^(\\d).*(\\d).*(\\d).*$" =>
+          { match = "5 apples, 6 pears and 0 grapes", index = 0, groups = [ "5", "6", "0" ] }
+        match "01234" "3" =>
+          { match = "3", index = 3, groups = [ ] }
+      ```
+      "#m
+    = fun s regex => %strMatch% s regex,
+
+    length : Str -> Num
+    | doc m#"
+      Results in the length of the given string.
+
+      For example:
+      ```nickel
+        length "" =>
+          0
+        length "hi" =>
+          2
+      ```
+      "#m
+    = fun s => %strLength% s,
+
+    substring: Str -> Num -> Num -> Str
+    | doc m#"
+      Takes a slice from the string. Errors if either index is out of range.
+
+      For example:
+      ```nickel
+        substring "abcdef" 3 5 =>
+          "de"
+        substring "abcdef" 3 10 =>
+          error
+        substring "abcdef" (-3) 4 =>
+          error
+      ```
+      "#m
+    = fun s start end => %strSubstr% s start end,
 
-    length : Str -> Num = fun s => %strLength% s,
+    fromNum | Num -> Str
+    | doc m#"
+      Converts a number to its string representation.
 
-    substring: Str -> Num -> Num -> Str = fun s start end =>
-      %strSubstr% s start end,
+      fromNum 42 =>
+        "42"
+      "#m
+    = fun n => %toStr% n,
 
-    fromNum | Num -> Str = fun n => %toStr% n,
     // fromEnum | < | Dyn> -> Str = fun tag => %toStr% tag,
-    fromEnum | Dyn -> Str = fun tag => %toStr% tag,
-    fromBool | Bool -> Str = fun b => %toStr% b,
+    fromEnum | Dyn -> Str
+    | doc m#"
+      Converts any basic value to its string representation.
+
+      For example:
+      ```nickel
+      fromEnum `MyEnum =>
+        "MyEnum"
+      ```
+      "#m
+    = fun tag => %toStr% tag,
+
+    fromBool | Bool -> Str
+    | doc m#"
+      Converts a boolean value to its string representation.
+
+      For example:
+      ```nickel
+        fromBool true =>
+          "true"
+      ```
+      "#m
+    = fun b => %toStr% b,
+
+    toNum | #NumLiteral -> Num
+    | doc m#"
+      Convers a string that represents an integer to that integer.
+
+      For example:
+      ```nickel
+        toNum "123" =>
+          123
+      ```
+      "#m
+    = fun s => %numFromStr% s,
+
+    toBool | #BoolLiteral -> Bool
+    | doc m#"
+      Converts a string that represents a bool to that bool.
+
+      For example:
+      ```nickel
+        toBool "true" =>
+          true
+        toBool "True" =>
+          true
+        toBool "false" =>
+          false
+      "#m
+    = fun s => s == "true",
 
-    toNum | #NumLiteral -> Num = fun s => %numFromStr% s,
-    toBool | #BoolLiteral -> Bool = fun s => s == "true",
     // toEnum | #Ident -> < | Dyn> = fun s => %enumFromStr% s,
-    toEnum | #Ident -> Dyn = fun s => %enumFromStr% s,
+    toEnum | #Ident -> Dyn
+    | doc m#"
+      Converts any string that represents an identifier to an enum variant.
```

Comment:
```suggestion
      Converts any string that represents an enum variant to that enum variant.
```

The thing is that valid enum variants and valid identifiers are the same thing. I agree it's not clear to have this contract named `Ident`. Maybe we should rename it to `EnumVariant`, `Enum` or `EnumTag` as it seems to be actually used as such.